packages/hebao_v1/contracts/stores/QuotaStore.sol

```diff
@@ -87,9 +71,11 @@ contract QuotaStore is DataStore, Claimable
         public
         onlyWalletModule(wallet)
     {
-        Quota memory q = quotas[wallet];
-        require(_hasEnoughQuota(q, amount), "QUOTA_EXCEEDED");
-        _addToSpent(wallet, q, amount);
+        uint available = availableQuota(wallet);
+        if (available != MAX_QUOTA) {
+            require(available >= amount, "QUOTA_EXCEEDED");
+            addToSpent(wallet, amount);
```

Comment:
This won't work, even when using `requireSelfOrWalletModule` on `addToSpent`. Changing this to `this.addToSpent(wallet, amount)` will work, but this will use ~1,700 gas more gas because it needs to do an external call.

Reply:
Will do!

Reply:
This actually does work *because* `msg.sender` remains the same... But we still don't want to do that because it's expensive to do the module check twice.